data/glossary.toml

```diff
@@ -0,0 +1,175 @@
+[[entries]]
+term = "Component"
+description = "A Pulumi component is a logical group of resources that contains other components and physical cloud resources."
+link = "/docs/intro/concepts/programming-model#components"
+
+[[entries]]
+term = "Stack"
+description = "A stack is an isolated, independently configurable instance of a Pulumi program. Stacks are commonly used to denote different phases of development (such as development, staging and production) or feature branches (such as feature-x-dev, jane-feature-x-dev)."
+link = "/docs/intro/concepts/stack"
+
+[[entries]]
+term = "Program"
+description = "Pulumi programs are authored in general-purpose programming languages such as JavaScript or Python. You can use any packages supported by the language’s package manager, as well as Pulumi packages."
+link = "/docs/intro/concepts/programming-model#programs"
+
+[[entries]]
+term = "Deployment Engine"
+description = "The deployment engine is responsible for computing the set of operations needed to drive the current state of your infrastructure into the desired state expressed by your program."
+link = "/docs/intro/concepts/how-pulumi-works"
+
+[[entries]]
+term = "Self-hosted"
+description = "Self-hosted in Pulumi applies to on-premise \"behind a firewall\" scenarios, as well as environments hosted within your own AWS, Azure, or GCP account."
+link = "/docs/intro/concepts/state#self-managed-backend"
+
+[[entries]]
+term = "Pulumi Console"
+description = "The Pulumi Console refers to the web application at `app.pulumi.com` which automatically manages deployment state and enables collaboration between developers and operators."
+link = "/docs/intro/console"
+
+[[entries]]
+term = "IdP"
+description = "IdP stands for Identity Provider. A Security Assertion Markup Language (SAML) IdP is a service that acts as a user directory."
+link = "https://en.wikipedia.org/wiki/Identity_provider#SAML_identity_provider"
+
+[[entries]]
+term = "SP"
+description = "SP stands for Service Provider. A Security Assertion Markup Language (SAML) service provider relies on an identity provider for authentication."
+link = "https://en.wikipedia.org/wiki/Service_provider_(SAML)"
+
+[[entries]]
+term = "IdP"
```

Comment:
Should this be "IdP Metadata XML"? There's already an "IdP" term on line 32.

Reply:
Yes, I've changed in my last commit. And the 3-column layout was exactly what I was envisioning. Thanks for adding a commit! And please approve if all looks good, so we can have this merged. :)